data/src/main/java/com/linkedin/data/schema/resolver/ClasspathResourceDataSchemaResolver.java

```diff
@@ -33,61 +34,106 @@
  *
  * @author Min Chen
  */
-public class ClasspathResourceDataSchemaResolver extends DefaultDataSchemaResolver
+public class ClasspathResourceDataSchemaResolver extends AbstractMultiFormatDataSchemaResolver
 {
   /**
    * The default file name extension is ".pdsc".
+   * @deprecated Do not use.
    */
+  @Deprecated
   public static final String DEFAULT_EXTENSION = SchemaParser.FILE_EXTENSION;
 
+  private final ClassLoader _classLoader;
+
+
   /**
    * Construct a new instance that uses the {@link Thread#getContextClassLoader()} for the current thread.
    */
+  public ClasspathResourceDataSchemaResolver()
+  {
+    this(null, Thread.currentThread().getContextClassLoader());
+  }
+
+  /**
+   * Construct a new instance that uses the {@link Thread#getContextClassLoader()} for the current thread.
+   *
+   * @deprecated The parserFactory is not needed as this class now uses builtin parsers. Use
+   * {@link #ClasspathResourceDataSchemaResolver()} instead
+   */
+  @Deprecated
   public ClasspathResourceDataSchemaResolver(SchemaParserFactory parserFactory)
   {
-    super(parserFactory);
-    _classLoader = Thread.currentThread().getContextClassLoader();
+    this(Thread.currentThread().getContextClassLoader());
   }
 
   /**
    * Construct a new instance that uses the specified {@link ClassLoader}.
    *
    * @param classLoader provides the {@link ClassLoader}.
    */
-  public ClasspathResourceDataSchemaResolver(SchemaParserFactory parserFactory, ClassLoader classLoader)
+  public ClasspathResourceDataSchemaResolver(ClassLoader classLoader)
   {
-    super(parserFactory);
+    for (DataSchemaParserFactory parserForFormat: BUILTIN_FORMAT_PARSER_FACTORIES)
+    {
+      SingleFormatClasspathSchemaResolver resolver = new SingleFormatClasspathSchemaResolver(parserForFormat);
+      resolver.setExtension("." + parserForFormat.getLanguageExtension());
```

Comment:
why do we need this setter exposed? Should constructor handle this automatically?

Reply:
+1

Reply:
Fixed, this was remnant of an alternate implementation I started with. Cleaned it now.